packages/sync-service/lib/electric/shape_cache/file_storage.ex

```
@@ -695,28 +696,8 @@ defmodule Electric.ShapeCache.FileStorage do
 
   @impl Electric.ShapeCache.Storage
   def unsafe_cleanup!(%FS{} = opts) do
-    unsafe_cleanup_with_retries!(opts)
-  end
-
-  defp unsafe_cleanup_with_retries!(%FS{} = opts, attempts_left \\ 5) do
-    with {:ok, _} <- File.rm_rf(opts.snapshot_dir),
-         {:ok, _} <- File.rm_rf(shape_definition_path(opts)),
-         {:ok, _} <- File.rm_rf(opts.data_dir) do
-      :ok
-    else
-      # There is a very unlikely but observed scenario where the rm_rf call
-      # tries to delete a directory after having deleted all its files, but
-      # due to some FS race the deletion fails with EEXIST. Very hard to test
-      # and prevent so we mitigate it with arbitrary retries.
-      {:error, :eexist, _} when attempts_left > 0 ->
-        unsafe_cleanup_with_retries!(opts, attempts_left - 1)
-
-      {:error, reason, path} ->
-        raise File.Error,
-          reason: reason,
-          path: path,
-          action: "remove files and directories recursively from"
-    end
+    {:ok, _} = File.rm_rf(opts.data_dir)
```

is there a reason this gets rid of the retries? the EEXIST error occurred on EFS with an rm_rf call which is why that was put in place, what was the reason for removing it?

---

i want to assert that the rm_rf is now working without issue now we're waiting for process termination before deleting things. i don't want to hide that error because it shows that something is not working  -- the retry was a hack, this pr is a fix, or at least it should be

---

right - but did you manage to reproduce that error before with your testing and then resolve it with your changes? cause it most likely was FS related rather than process related - `rm_rf` should never raise `EEXIST` so it's more like a freak error. Even if the process is dead the FS might not have flushed things and we might still run into it.

---

after all that I've restored the retry as the :eexist issue is happening in tests.